tephra-hbase-compat-0.94/src/main/java/co/cask/tephra/hbase94/TransactionAwareHTable.java

```diff
@@ -515,6 +506,46 @@ private ActionChange(byte[] row, byte[] family, byte[] qualifier) {
     private byte[] getQualifier() {
       return qualifier;
     }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
Good idea.  On looking at this again, I think we can actually extract a common base class for the TransactionAwareHTable implementations, so we can reduce a lot of the code duplication here.